ui-modules/blueprint-composer/app/components/catalog-saver/catalog-saver.directive.js

```diff
@@ -162,9 +163,15 @@ export function CatalogItemModalController($scope, blueprintService, paletteApi,
 
         let bomItem = {
             id: $scope.config.symbolicName,
-            itemType: $scope.config.itemType,
             item: blueprint
         };
+
+        // Set the itemType to the correct value or leave out entirely if config undefined
+        let itemType = $scope.config.entity ? 'entity' : $scope.config.template ? 'template' : undefined;
```

Comment:
this logic is not what the use of checkboxes implies.  also, if nothing is selected `undefined` will generate invalild catalog BOM, no?

suggest leaving it as radiobuttons with a variable `$scope.config.itemType` defaulting to `template ?